spec/datadog/core/configuration/stable_config_spec.rb

```diff
@@ -12,4 +14,93 @@
       end
     end
   end
+
+  describe '#configuration', skip: !LibdatadogHelpers.supported? do
+    let(:tmpdir) { Dir.mktmpdir }
+    before do
+      FileUtils.mkdir_p(File.join(tmpdir, 'managed/datadog-agent/stable'))
+      # local config
+      File.write(
+        File.join(tmpdir, 'application_monitoring.yaml'),
+        local_config_content
+      )
+      # fleet config
+      File.write(
+        File.join(tmpdir, 'managed/datadog-agent/stable/application_monitoring.yaml'),
+        fleet_config_content
+      )
```

Comment:
Minor: The local config / fleet config are already written in code via the names of the lets ;)

```suggestion
      File.write(
        File.join(tmpdir, 'application_monitoring.yaml'),
        local_config_content
      )
      File.write(
        File.join(tmpdir, 'managed/datadog-agent/stable/application_monitoring.yaml'),
        fleet_config_content
      )
```